src/System.Linq.Parallel/tests/PlinqModesTests.cs

```diff
@@ -11,157 +11,135 @@ namespace Test
+                    foreach (ParallelExecutionMode mode in modes)
+                    {
+                        yield return new object[] { results[0], results[1], query, mode };
+                    }
                 }
             }
         }
 
-        #region Helper Methods / Classes
+        [Theory]
+        [OuterLoop]
+        [MemberData("EasyQueryData", new[] { 1024 }, new[] { ParallelExecutionMode.Default, ParallelExecutionMode.ForceParallelism })]
+        [MemberData("HardQueryData", new[] { 1024 }, new[] { ParallelExecutionMode.ForceParallelism })]
+        public static void WithExecutionMode(Labeled<ParallelQuery<int>> labeled, int count, Labeled<Action<ParVerifier, ParallelQuery<int>>> operation, ParallelExecutionMode mode)
+        {
+            ParVerifier verifier = new ParVerifier();
+            operation.Item(verifier, labeled.Item.WithExecutionMode(mode));
+            verifier.AssertPassed();
+        }
 
-        private static void Iterate<T>(this IEnumerable<T> e)
+        [Theory]
+        [MemberData("Ranges", (object)(new int[] { 2 }), MemberType = typeof(UnorderedSources))]
+        public static void WithExecutionMode_ArgumentException(Labeled<ParallelQuery<int>> labeled, int count)
         {
-            foreach (var x in e) { }
+            ParallelQuery<int> query = labeled.Item;
+
+            Assert.Throws<ArgumentException>(() => query.WithExecutionMode((ParallelExecutionMode)2));
         }
 
-        // A class that checks whether Verify has been called from one or multiple threads.
-        private abstract class Verifier
+        [Fact]
+        public static void WithExecutionMode_ArgumentNullException()
         {
-            internal abstract int Verify(int x);
+            Assert.Throws<ArgumentNullException>(() => ((ParallelQuery<int>)null).WithExecutionMode(ParallelExecutionMode.Default));
+        }
 
-            internal abstract bool Passed { get; }
+        private static void ForEach<T>(this IEnumerable<T> e)
+        {
+            foreach (var x in e) { }
         }
 
         // A class that checks whether the Verify method got called from at least two threads.
         // The first call to Verify() blocks. If another call to Verify() occurs prior to the timeout
         // then we know that Verify() is getting called from multiple threads.
-        private class ParVerifier : Verifier
+        public class ParVerifier
         {
             private int _counter = 0;
             private bool _passed = false;
-            private const int TIMEOUT_LIMIT = 30000;
 
-            internal override int Verify(int x)
+            internal int Verify(int x)
             {
                 lock (this)
                 {
                     _counter++;
                     if (_counter == 1)
                     {
-                        if (Monitor.Wait(this, TIMEOUT_LIMIT))
```

Comment:
"Because it offended my sensibilities".  Probably much better to fail the test than hang, yes, changing back.